api/src/auth/auth.ts

```diff
@@ -29,54 +29,36 @@ export abstract class AuthDriver {
 	abstract verify(user: User, password?: string): void;
 
 	/**
-	 * Handle user session refresh
+	 * Check with the (external) provider if the user is allowed entry to Directus
 	 *
-	 * @param user User information
-	 * @param session Session data
+	 * @param _user User information
+	 * @param _payload Any data that the user might've provided
 	 * @throws InvalidCredentialsException
 	 */
-	refresh(_user: User, _session: Session): void {
-		/* Optional */
+	async login(_user: User, _payload: Record<string, any>): Promise<Record<string, any> | null> {
```

Comment:
It would be nice to make `Record<string, any> | null` it's own type `SessionData`.